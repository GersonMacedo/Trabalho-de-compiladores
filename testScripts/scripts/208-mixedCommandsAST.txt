! -s TREE
! should print correctly the AST with the if else commands
program p ;
    var b1 : boolean;
    var b2 : boolean;
    var b3 : boolean;
    var i1 : integer;
    var i2 : integer;
    var i3 : integer;
    var i4 : integer;
begin
    if b1 then
        while b2 do
        begin
            i1 := i2;
            i3 := i4;
            if b2 then
                while b1 do
                    b2 := false
            else
                while b2 do
                    b1 := true
            ;
        end
    else
        while b3 do
        begin
            i2 := i1;
            i4 := i3;
            if b1 then
                while b2 do
                    b1 := true
            else
                while b1 do
                    b2 := false
            ;
        end
    ;
    begin
        i3 := 4;
        i4 := 3;
    end ;
end .