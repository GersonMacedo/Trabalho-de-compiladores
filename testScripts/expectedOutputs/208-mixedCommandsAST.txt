
Performing up to syntactic analysis and printing the AST tree at testScripts/scripts/208-mixedCommandsAST.txt
No errors found

Programa 'p'
Declarações:
    'b1': BOOLEAN
    'b2': BOOLEAN
    'b3': BOOLEAN
    'i1': INTEGER
    'i2': INTEGER
    'i3': INTEGER
    'i4': INTEGER
Corpo:
    if
        'b1'
    then
        while
            'b2'
        do
            'i1' :=
                'i2'
            'i3' :=
                'i4'
            if
                'b2'
            then
                while
                    'b1'
                do
                    'b2' :=
                        false
            else
                while
                    'b2'
                do
                    'b1' :=
                        true
    else
        while
            'b3'
        do
            'i2' :=
                'i1'
            'i4' :=
                'i3'
            if
                'b1'
            then
                while
                    'b2'
                do
                    'b1' :=
                        true
            else
                while
                    'b1'
                do
                    'b2' :=
                        false
    'i3' :=
        4
    'i4' :=
        3
